src/cli.c

```diff
@@ -0,0 +1,115 @@
+#include "cli.h"
+#include "logging.h"
+
+#include <stdlib.h>
+#include <string.h>
+#include <unistd.h>
+#include <limits.h>
+#include <getopt.h>
+#include <stdio.h>
+#include <errno.h>
+
+/******************************************************
+ *** LOCAL DEFINES                                  ***
+ ******************************************************/
+
+#define MAX_HZ (1000000L)
+
+/******************************************************
+ *** LOCAL VARIABLES                                ***
+ ******************************************************/
+
+static const char* usage_str = "yobemag [-d <0..4>] <ROM>";
+
+/******************************************************
+ *** LOCAL METHODS                                  ***
+ ******************************************************/
+
+static int safe_strtol(const char* str_to_conv, int* store_into) {
+    char* end;
+    const long strtol_in = strtol(str_to_conv, &end, 10);
+
+    errno = 0;
+
+    if (end == str_to_conv) {
+        LOG_FATAL("%s: not a decimal number", str_to_conv);
+    } else if ('\0' != *end) {
+        LOG_FATAL("%s: extra characters at end of input: %s", str_to_conv, end);
+    } else if ((LONG_MIN == strtol_in || LONG_MAX == strtol_in) && ERANGE == errno) {
+        LOG_FATAL("%s out of range of type long", str_to_conv);
+    } else if (strtol_in > INT_MAX) {
+        LOG_FATAL("%ld greater than INT_MAX", strtol_in);
+    } else if (strtol_in < INT_MIN) {
+        LOG_FATAL("%ld less than INT_MIN", strtol_in);
+    } else {
+        *store_into = (int)strtol_in;
+        return EXIT_SUCCESS;
+    }
+
+    return EXIT_FAILURE;
+}
+
+/******************************************************
+ *** EXPOSED METHODS                                ***
+ ******************************************************/
+
+CLIArguments* cli_config_default(void) {
+    CLIArguments* conf = malloc(sizeof(CLIArguments));
+
+    conf->debug = FATAL;
+    conf->us_delay = 0;
+    conf->rom_path = "";
+
+    return conf;
+}
+
+void cli_config_destroy(CLIArguments *conf) {
+    free(conf);
+}
+
+int cli_config_handle(CLIArguments* const conf, int argc, char **argv) {
+    int c;
+
+    if (argc < 2) {
+        LOG_FATAL("No ROM path specified!");
+        fprintf(stderr, "%s\n", usage_str);
+        return EXIT_FAILURE;
+    }
+
+    // parse all options first
+    int strtol_in;
+    while ((c = getopt(argc, argv, "d:")) != -1) {
```

Comment:
Maybe we should change the name of this flag to be `l` or something that indicates log level instead of "debug"

Reply:
That's fair